spec/controllers/rooms_controller_spec.rb

```diff
@@ -173,6 +173,26 @@
       expect(response).to have_http_status(:forbidden)
     end
 
+    context 'the access code meeting option is enabled' do
+      let(:fake_room_settings_getter) { instance_double(RoomSettingsGetter) }
+      let!(:meeting_option) { create(:meeting_option, name: 'glViewerAccessCode') }
+      let!(:room_meeting_option) { create(:room_meeting_option, meeting_option:) }
+
+      before do
+        allow(RoomSettingsGetter).to receive(:new).and_return(fake_room_settings_getter)
+        allow(fake_room_settings_getter).to receive(:call).and_return(
+          { 'glViewerAccessCode' => true, 'glModeratorAccessCode' => false }
+        )
+        # can we avoid hardcoding the friendly id?
+        allow(SecureRandom).to receive(:alphanumeric).and_return('8ac31fcc-b550-48f3-a1cd-eab7ec413f35', 'abc123')
```

Comment:
You don't need this check - you can just check that it's not blank below

Reply:
The RoomMeetingOption factory is doing `value { [true, false, '12345'].sample }`
I will see if I can play with it without breaking the other specs.

Reply:
I'm not sure what you mean. The value in the db is not relevant since you're stubbing the RoomSettingsGetter response anyways.

Reply:
Maybe there is something I didn't get, but the RoomSettingsGetter returns a true or false depending if the access_code option is enabled or not. 

If I want to get the actual value of the access_code, the string value, I have to get it elsewhere.

In the specs, and in the room#generate_access_code, I will find the value by matching the RoomMeetingOption/MeetingOption with the room id and the name of the meeting option.